packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -91,20 +71,26 @@ export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
       }
 
       // Recursively collect `.tsx` files in the given directory
-      const collectFiles = async (dir: string): Promise<string[]> => {
-        if (!pagesDir) return [];
-        const results: string[] = [];
-        const files = await readdir(dir, {
-          withFileTypes: true,
-          recursive: true,
-        });
-
-        for (const file of files) {
-          if (file.name.endsWith('.tsx')) {
-            results.push('/' + file.name);
+      const collectFiles = async (
+        dir: string,
+        cwd: string = '',
+        files: string[] = [],
+      ): Promise<string[]> => {
+        if (!cwd) {
+          cwd = dir;
```

Comment:
Is cwd needed? each entry from readDir has it's parent path

Reply:
dir is the starting dir and cwd is the current dir as it recurses. It needs the starting dir to create the relative path. It probably could use `pagesDir` from the outer scope though instead of passing around in the function.

Reply:
pagesDir seems good to re-use. and I am seeing now that entry.parentPath is only available in node 20, so nvm about that.

Reply:
Yeah, they started with `parentPath` and then changed it to just `path`. I tried to use too when generating the _routes.json file for the Cloudflare Pages build and found those issues with it and pivoted to a recursive function - so it was easy for me to apply that here.